src/gui/dialogs/logging.cpp

```diff
@@ -63,7 +63,8 @@ void tlogging::pre_show(twindow& window)
 		std::map<std::string, string_map> data;
 		string_map item;
 
-		for(unsigned int iter = (widget_id_.size()-1); iter < widget_id_.size(); iter--){
+		unsigned int id_size = widget_id_.size();
+		for(unsigned int iter = (id_size-1); iter < id_size; iter--){
```

Comment:
This loop looks like it's asking for problems. It starts at 3, continues as long as `iter` is less than 4, and decrements the counter each time. Also, the name `iter` is misleading - it's a counter, not an iterator, so you should call it `i` or `j` or `n` or `m` or `counter` or something.